src/isomorphic/classic/types/__tests__/ReactPropTypes-test.js

```diff
@@ -40,7 +43,9 @@ function typeCheckPass(declaration, value) {
     props,
     'testProp',
     'testComponent',
-    ReactPropTypeLocations.prop
+    ReactPropTypeLocations.prop,
+    null,
+    ReactPropTypesSecret
```

Comment:
> I’m a bit concerned about this. We’re not actually testing that React is passing the secret when validating.

There is a test in `ReactPropTypesProduction-test.js` that makes sure the warnings aren't being called when prop types are validated internally.

> So if React doesn’t do it in some case.

React should do it all cases, I updated all the files that called a PropType function internally, and I would say it's reasonable that any future changes that involve calling one of these PropType functions should also be sure to pass in `ReactPropTypeSecretValue`. 

> we’ll miss this.

It's hard to miss, the test suite blows up if a warning is logged when it shouldn't be.

> Can we change these typeCheckFail and typeCheckPass to be more like integration tests? i.e. they would create a class and then make sure that calling createElement produces (or doesn’t produce) the relevant warning. Or am I missing something?

I can definitely do that. As is I don't think it makes a difference, but it would ensure that future changes to how PropTypes are checked don't require this test to be updated as well.

The only issue I see is that if we do it this way we can't get the actual error instance returned from `declaration` so it would have to rely on spying on `console.error` instead.